src/debug/di/shimstackwalk.cpp

```diff
@@ -1663,6 +1663,11 @@ HRESULT ShimChain::GetStackRange(CORDB_ADDRESS * pStart, CORDB_ADDRESS * pEnd)
         {
             *pEnd = PTR_TO_CORDB_ADDRESS(m_fpRoot.GetSPValue());
         }
+
+        if (m_fpRoot.GetSPValue() < CORDbgGetSP(&m_context))
```

Comment:
Should we also consider the case where m_fpRoot.GetSPValue() == CORDbgGetSP()? In that case the stack unwinder is making no progress too.

Reply:
The stronger check is probably valid, but I'd feel a little more nervous about adding it. Without doing some extra investigation its hard for me to rule out that we'd never have a situation with a zero sized chain that still does make progress overall (ie not all the chains are zero sized).